src/repositories/user.repository.ts

```diff
@@ -0,0 +1,27 @@
+import {IUserRepository} from "./user.repository.interface";
+import {Repository} from "typeorm/repository/Repository";
+import {User} from "../entity/user.entity";
+import {v4 as uuidv4} from "uuid";
+import {dataSource} from "../dataSource";
+
+export class UserRepository implements IUserRepository {
+    constructor(private repository: Repository<User>) {}
+
+   async getUserByEmail(email: string): Promise<User> {
+        return await this.repository.findOne({ where: { email } })
+    }
+
+    createUser(id: string, email: string): User {
+        return this.repository.create({ id: uuidv4(), email })
+    }
+
+    async saveUser(user: User): Promise<void> {
```

Comment:
as far as I know it typeORM you need to create record and then save it to add to the database so in your custom repository you can combine these two methods and create general one that will perform creating and saving